net.solarnetwork.node.setup.web/web/js/graph-data.js

```diff
@@ -0,0 +1,237 @@
+                //and have 1 tick
+                if (data[0] == 0){
+                    //the +-10 were arbitrary just needed 0 in the middle
+                    ydomain = [-10,10];
+
+                    //the 1 tick will appear on the only datapoint which is 0
+                    yticks = 1;
+                }else{
+
+                    //the number could be positive or negative, (not tested when data is 0)
+                    ydomain = [Math.min(0, data[0]), Math.max(0, data[0])];
+
+                    //slightly extend the domain so the datapoint is not right at the edge of the graph
+                    //1.1 was chosen for aesthetics 
+                    ydomain[0] = ydomain[0] * 1.1;
+                    ydomain[1] = ydomain[1] * 1.1;
+
+                    //the 3 ticks will show the domain range and the 1 datapoint
+                    yticks = 3;
+                }           
+                
+            } else {
+
+                //note not true standard dev of the data but gives a nice look
+                //take the stardard dev of the unique values to get a get a nice whitespace above and below
+                //the graph
+                var stddev = StandardDeviation(unique);
+
+                //extends the domain using the calculated stddev
+                ydomain = [d3.min(data) - stddev, d3.max(data) + stddev];
+
+                //5 ticks was chosen aesthetically, not too many to cause crowding on the graph
+                //not too few to stop reading the graph
+                yticks = 5;
+            }
+            
+            //applies the new domain
+            y.domain(ydomain)
+
+            //sets the new ticks
+            svg.select("g .yaxis")
+                //the number of ticks is the minimum of the number of unique values and 5 
+                .call(d3.svg.axis().scale(y).ticks(yticks).orient("left"));
+
+            //grabs the datapoint and puts it in the data array
+            data.push(datamap[source]);
+
+            // redraw the line
+            svg.select(".line")
+                .attr("d", line)
+                .attr("transform", null);
+
+            // slide the x-axis left
+            axis.call(x.axis);
+
+            // slide the line left
+            path.transition()
+                .attr("transform", "translate(" + x(now - (n - 1) * duration) + ")");
+
+            // pop the old data point off the front
+            data.shift();
+
+            //hack to only show 2dp accuracy probably a better way but couldn't find it
+            svg.selectAll("g .yaxis").selectAll("g .tick").selectAll("text").each(function roundtick(d, i) {
+                ticktext = d3.select(this).text();
+
+                //this regex checks for positive and negative numbers that can have commas , and 0-2 decimal places
+                //NOTE this could break depending on the locale numbers are formatted in
+                var re = new RegExp('-?([0-9]|,)+(\.[0-9][0-9]?)?');
```

Comment:
I'd move the `re` variable to a top-level class variable so the expression doesn't have to be parsed on every tick.